File: Organizing Data C#/DesignPatternGK/SelfEncapsulateField/Rectangle.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SelfEncapsulateField
{
    class Rectangle
    {
        private double _length, _width;

        public Rectangle(double length, double width)
        {
            _length = length;
            _width = width;
        }
        public double Length
        {
            set { _length = value; }
            get { return _length; }
        }
        public double Width
        {
            set { _width = value; }
            get { return _width; }
        }

        public double getArea()
        {
            return (Length * Width);
        }

        public double getPerimeter()
        {
            return (2 * Length+ Width);
        }
    }
}

File: Organizing Data Java/src/tdtu/edu/OrganizingData/SelfEncapsulateField/Rectangle.java
package tdtu.edu.OrganizingData.SelfEncapsulateField;

public class Rectangle {
	private double length, width;

	public Rectangle(double length, double width) {
		super();
		initialize(length,width);
	}

	private void initialize(double length, double width) {
		this.length = length;
		this.width = width;
	}

	public double getArea() {
		return (getLength() * getWidth());
	}

	public double getPerimeter() {
		return (2 * (getLength() + getWidth()));
	}

	public double getLength() {
		return length;
	}

	public void setLength(double length) {
		this.length = length;
	}

	public double getWidth() {
		return width;
	}

	public void setWidth(double width) {
		this.width = width;
	}
}
